cmd/xl-sets.go

```diff
@@ -823,11 +823,13 @@ func (f *FileInfoCh) Push(fi FileInfo) {
 
 // Calculate least entry across multiple FileInfo channels, additionally
 // returns a boolean to indicate if the caller needs to call again.
-func leastEntry(entriesCh []FileInfoCh, totalDrives int, heal bool) (FileInfo, bool) {
-	var entriesValid = make([]bool, len(entriesCh))
-	var entries = make([]FileInfo, len(entriesCh))
-	for i := range entriesCh {
-		entries[i], entriesValid[i] = entriesCh[i].Pop()
+func leastEntry(entryChs []FileInfoCh, heal bool, entries []FileInfo, entriesValid []bool) (FileInfo, int, bool) {
+	for i := range entryChs {
+		entries[i], entriesValid[i] = FileInfo{}, false
```

Comment:
Here you clear it, and overwrite it just below, or am I mistaking?

Reply:
Each function needs a new state for each entry in the slice.

Reply:
This for loop is not needed. @klauspost is right that, the slices are simply overwritten in the next for loop.

Reply:
Done, the previous code was a communicative code to indicate that we start with a clean slate. But just doing Pop() is fine too. 

Updated the PR.